### DISCUSSION TITLE
[Bug]: Ports positioning in jointJS with angular

### DISCUSSION THREAD

**Discussion by meyabishekssciera** [clientIO/joint#2466] [Q&A]
### What happened?

Hi all,

`       When I used JointJS with angular, the ports positioning is not supported in the paper ,I have given the output port position of the port to be right, but it is rendering in the left side of the element 
`

 ```
// Out Port
    var portsOut = {
      position: {
        name: 'right',
      },
      attrs: {
        portBody: {
          magnet: true,
          r: 6,
          fill: '#E6A502',
          stroke: '#023047',
        }
      },
      label: {
        position: {
          name: 'right',
          args: { x: 6 }  
        },
        markup: [{
          tagName: 'text',
            selector: 'label',
            className: 'label-text',
        }]
      },
      markup: [{
        tagName: 'circle',
        selector: 'portBody'
      }]
    };
```

<img width="618" alt="image" src="https://github.com/clientIO/joint/assets/125563279/e2e73397-bbc5-4aa2-8e9b-0fa7cf07173b">

### Version

3.7.7

### What browsers are you seeing the problem on?

Chrome,Firefox, Safari, Microsoft Edge

### What operating system are you seeing the problem on?

MacOS,Windows

**Comment by kumilingus** [comment]
You need to define the position for a group and assign the port to the group.

**Reply by meyabishekssciera** [reply]
```
        ports: {
          groups: {
            'in': {  
              position: { name: 'left' }  
            },
            'out': {  
              position: { name: 'right' } 
            }
          },
          items: ([portsIn, portsOut]) 
        }
```
        
`     I have already defined the position and have assigned the ports to the group, but still the output port  is not positioning in the right`
